### DISCUSSION TITLE
Display the user input from a survey-text response as the prompt for the next survey-text

### DISCUSSION THREAD

**Discussion by avashipman** [jspsych/jsPsych#798] [Q&A] [answered]
Hi,
I am pretty new to jsPsych and JavaScript/HTML, but I am trying to have an experiment where the user responds to the prompt in a textbox and then clicks a button where next their answer will be the prompt for another textbox below it. That would be 1 trial and would repeat with other questions for the next trials. I am not sure how to put a function in the prompt parameter for survey-text because it only reads string. I am just having a hard time finding a command that will display the user's answer as a prompt. Here is the code I have so far:

var noisy_channel = {
    timeline: [
        {
            type: 'survey-text',
            questions: [{prompt: jsPsych.timelineVariable('prompt')},
            {prompt: JSON.parse(jsPsych.data.get().filter( {trial_type: 'survey-text'} ).select('responses').values).prompt,},
            ],
            preamble: "Please write one sentence.",
            button_label:"SUBMIT"
        },
    ],
    timeline_variables: [
        { prompt: 'What did you do last weekend?' },
        { prompt: 'What is your favorite season and why'},
        { prompt: 'What do you think you’ll do on your next birthday?'},
        { prompt: 'How would you explain something complicated to a child?'}
    ],
    randomize_order: true
}

jsPsych.init({
        timeline: [noisy_channel],
        on_finish: function() {jsPsych.data.displayData()},
    })

Any help will be greatly appreciated!

**Comment by vijaymarupudi** [comment]
The key is to make the questions option a function which returns the array that you want.

When you make it a function, make sure you set the [second parameter of timelineVariable to true](https://www.jspsych.org/core_library/jspsych-core/#invoking-immediately-in-a-function).

**Reply by avashipman** [reply]
Ok, that makes sense, thank you! 

How would I make questions into a function that would show the prompt and then display their response so they can fill in another textbox based on their response? 

This didn't work but am I on the right track:

```js
questions:[ 
	{prompt: jsPsych.timelineVariable('prompt')},
        function( ) {
		return JSON.parse(jsPsych.data.getLastTrialData().select('responses').values).prompt,
             	src= +jsPsych.timelineVariable('prompt', true);
        }
],
```

**Comment by vijaymarupudi** [comment, marked as answer]
I'm not exactly sure if this is what you're trying to do?

```js
  questions: function () {
    const prevPrompt = JSON.parse(
      jsPsych.data.getLastTrialData().values()[0].responses
    ).prompt;

    return [
      { prompt: jsPsych.timelineVariable("prompt", true) },
      { prompt: prevPrompt },
    ];
  },
```

**Reply by avashipman** [reply]
Thank you for this work! 

I realized that what I am asking is a bit confusing. I also see that having a participant's response show up as the prompt might not work in the same trial. I want to retrieve the user's response so then it can become the next prompt for the user to see. This can be in the next trial so that there would be data in the first place. To clarify, this is what I would ideally want:
![IMG_6274](https://user-images.githubusercontent.com/52719718/86503996-5a1ae700-bd4f-11ea-8dbe-6ae98bdf6a75.jpg)

Does this make more sense? Thank you again for all of your help! I really appreciate it!

**Reply by vijaymarupudi** [reply]
I get it now, so this is probably what you want to do:

See <https://www.jspsych.org/overview/timeline/#timeline-variables> for reference.


```javascript
var one_item_block = {
  timeline: [
    {
      type: "survey-text",
      questions: function () {
        return [{ prompt: jsPsych.timelineVariable("prompt", true) }];
      },
    },
    {
      type: "survey-text",
      questions: function () {
        const prev_answer = JSON.parse(
          jsPsych.data.getLastTrialData().values()[0].responses
        ).Q0;

        var new_prompt = "<p>Please edit your sentence or copy it?</p>";
        new_prompt += "<p>" + prev_answer + "</p>";

        return [{ prompt: new_prompt }];
      },
    },
  ],
};

timeline.push(one_item_block)
```

**Reply by avashipman** [reply]
This worked! Thank you so so much! When it ran smoothly I teared up.

**Reply by vijaymarupudi** [reply]
I'm glad, you're welcome!
